src/java.base/share/classes/java/time/format/DateTimeFormatterBuilder.java

```diff
@@ -2488,15 +2561,130 @@ interface DateTimePrinterParser {
         int parse(DateTimeParseContext context, CharSequence text, int position);
     }
 
+    static final class LocalDatePrinterParser extends CompositePrinterParser {
+        LocalDatePrinterParser() {
+            super(buildPrinterParsers(), false);
+        }
+
+        private static List<DateTimePrinterParser> buildPrinterParsers() {
+            List<DateTimePrinterParser> printerParsers = new ArrayList<>();
+            printerParsers.add(new NumberPrinterParser(ChronoField.YEAR, 4, 10, SignStyle.EXCEEDS_PAD));
+            printerParsers.add(new CharLiteralPrinterParser('-'));
+            printerParsers.add(new NumberPrinterParser(ChronoField.MONTH_OF_YEAR, 2, 2, SignStyle.NOT_NEGATIVE));
+            printerParsers.add(new CharLiteralPrinterParser('-'));
+            printerParsers.add(new NumberPrinterParser(ChronoField.DAY_OF_MONTH, 2, 2, SignStyle.NOT_NEGATIVE));
+            return printerParsers;
+        }
+
+        @Override
+        public boolean format(DateTimePrintContext context, StringBuilder buf) {
+            TemporalAccessor temporal = context.getTemporal();
+
+            LocalDate date;
+            if (temporal instanceof LocalDateTime) {
+                date = ((LocalDateTime) temporal).toLocalDate();
+            } else if (temporal instanceof LocalDate) {
+                date = (LocalDate) temporal;
+            } else if (temporal instanceof ZonedDateTime) {
+                date = ((ZonedDateTime) temporal).toLocalDate();
+            } else if (temporal instanceof OffsetDateTime) {
+                date = ((OffsetDateTime) temporal).toLocalDate();
+            } else {
+                return super.format(context, buf);
+            }
```

Comment:
This can use `instanceof <pattern>`:
```suggestion
            if (temporal instanceof LocalDateTime ldt) {
                date = ldt.toLocalDate();
            } else if (temporal instanceof LocalDate ld) {
                date = ld;
            } else if (temporal instanceof ZonedDateTime zdt) {
                date = zdt.toLocalDate();
            } else if (temporal instanceof OffsetDateTime odt) {
                date = odt.toLocalDate();
            } else {
                return super.format(context, buf);
            }
```

Or even a pattern switch:
```suggestion
            switch (temporal) {
                case LocalDateTime ldt -> date = ldt.toLocalDate();
                case LocalDate ld -> date = ld;
                case ZonedDateTime zdt -> date = zdt.toLocalDate();
                case OffsetDateTime odt -> date = odt.toLocalDate();
                default -> {
                    return super.format(context, buf);
                }
            }
```